src/crimson/os/seastore/lba_mapping.h

```diff
@@ -0,0 +1,79 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include "crimson/os/seastore/cached_extent.h"
+#include "crimson/os/seastore/btree/btree_range_pin.h"
+
+namespace crimson::os::seastore {
+
+class LBAMapping;
+using LBAMappingRef = std::unique_ptr<LBAMapping>;
+
+class LogicalCachedExtent;
+
+class LBAMapping : public BtreeNodeMapping<laddr_t, paddr_t> {
+public:
+  LBAMapping(op_context_t<laddr_t> ctx)
+    : BtreeNodeMapping<laddr_t, paddr_t>(ctx) {}
+  template <typename... T>
+  LBAMapping(T&&... t)
+    : BtreeNodeMapping<laddr_t, paddr_t>(std::forward<T>(t)...)
+  {
+    if (!is_parent_pending()) {
+      this->child_pos = {parent, pos};
+    }
+  }
+
+  // An lba pin may be indirect, see comments in lba_manager/btree/btree_lba_manager.h
+  virtual bool is_indirect() const { return false; }
+  virtual laddr_t get_intermediate_key() const {
+    return min_max_t<laddr_t>::null;
+  }
+  virtual laddr_t get_intermediate_base() const {
+    return min_max_t<laddr_t>::null;
+  }
+  virtual extent_len_t get_intermediate_length() const { return 0; }
+  // The start offset of the pin, must be 0 if the pin is not indirect
+  virtual extent_len_t get_intermediate_offset() const {
+    return std::numeric_limits<extent_len_t>::max();
```

Comment:
Are these default implementations for virtual methods still meaningful?

I suppose they were introduced because backref mapping needed dummy implementation?

Reply:
Ah, yes, the default implementations should be removed:-) Will do.